extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -1751,11 +1761,24 @@ export class AmpStoryPage extends AMP.BaseElement {
       return;
     }
 
+    // Calculating contrast color (black or white) needed for outlink CTA UI.
+    const ctaAccentColor = attachmentEl.getAttribute('cta-accent-color');
+    let contrastColor = null;
+    if (ctaAccentColor) {
+      setImportantStyles(this.element, {
+        'background-color': attachmentEl.getAttribute('cta-accent-color'),
+      });
+      const styles = computedStyle(this.win, this.element);
+      const rgb = getRGBFromCssColorValue(styles['background-color']);
+      contrastColor = getTextColorForRGB(rgb);
+    }
```

Comment:
This code should live in the page-open-attachment file (inside the `if (theme=custom)`), it's related to the rendering of the outlink UI and not the page